ユーザー情報編集機能の追加

Kaoru1126 opened the pull request:
# what
ユーザーの編集機能の追加
-名前とメールアドレスの変更


# why
ChatSpaceの開発に必要だから

Kaoru1126 committed:
left commiting from PR#5

did not submit 2 commits from last PR.

Kaoru1126 committed:
taking memo

not sure if i should do it now.

Kaoru1126 committed:
generated "users_controller"

Kaoru1126 committed:
added edit, update, user_params, and redirect methods.

Kaoru1126 committed:
added spaces

Kaoru1126 committed:
added this file from DL

Kaoru1126 committed:
changed "before action..."

move_to_index, :except => [:index] => authenticate_user!

Kaoru1126 committed:
added path

Kaoru1126 committed:
made icons linkable

Kaoru1126 committed:
added some lines to scss "_sidebar"

for icon_gear and _pen

Kaoru1126 committed:
made icons clickable with "link_to"

and erased some texts in scss for those icons

Kaoru1126 committed:
added action after editing the user info

and when user could not change its their info

Kaoru1126 committed:
added gem "pry-rails"

Kaoru1126 committed:
added view files from DL files

Kaoru1126 committed:
before taking action

Kaoru1126 committed:
revived files with user.rb

Kaoru1126 committed:
solving errors after revived files

Kaoru1126 committed:
added variable

there is an error saying  undefined variables for $white-text, so i
added the text

Kaoru1126 committed:
to logging_out

Kaoru1126 committed:
to give id info when link to edit

Kaoru1126 committed:
to get user info that's gonna replace for original

Kaoru1126 committed:
deleted extra lines I wrote accidentally

seoky0 commented on app/assets/javascripts/users.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
消しましょう

Kaoru1126 replied:
ファイルごと削除しました

seoky0 commented on app/assets/stylesheets/_sidebar.scss:
```diff
@@ -1,3 +1,5 @@
+@import "./variables";
```
この記述はapplication.scssで読み込みましょう

Kaoru1126 replied:
application.scss内にも同じ記述があるのですが、
_sidebar.scssから消すと、以下のエラーがでます。

Sass::SyntaxError in Users#edit
Undefined variable: "$white_text".

Kaoru1126 replied:
解決しました。修正済みです。

seoky0 commented on app/assets/stylesheets/users.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the users controller here.
```
消しましょう

Kaoru1126 replied:
ファイルごと削除しました

seoky0 commented on app/controllers/messages_controller.rb:
```diff
@@ -1,5 +1,25 @@
 class MessagesController < ApplicationController
 
+  def edit
+    @user = User.find(params[:id])
```
updateと同じ記述をしているので、メソッドを定義してそのメソッドをbefore_actionで呼び出しましょう

seoky0 replied:
すみません。
この記述が全てmessagesコントローラなのでuserを更新する記述は全て消しましょう。

Kaoru1126 replied:
全部けしました。

seoky0 commented on app/controllers/messages_controller.rb:
```diff
@@ -1,5 +1,25 @@
+      params.require(:user).permit(:name, :email)
+  end
+
+  def move_to_index
```
この記述は不要ですね。
ログインしていない人はルートにもアクセスできません。
ログインしていない人は一律ログインページに飛ばしましょう

Kaoru1126 replied:
リダイレクトを追加しました

seoky0 commented on app/helpers/users_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
```
けしましょう

Kaoru1126 replied:
ファイルごと消しました。

seoky0 commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,24 @@
+before_action :move_to_index, :except => [:index]
+
+  def edit
+    @user = User.find(params[:id])
```
updateと同じ記述をしているので、メソッドを定義してそのメソッドをbefore_actionで呼び出しましょう

Kaoru1126 replied:
修正済みです。

seoky0 commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,24 @@
+      params.require(:user).permit(:name, :email)
+  end
+
+  def move_to_index
```
この記述は不要ですね。
ログインしていない人はルートにもアクセスできません。
ログインしていない人は一律ログインページに飛ばしましょう

Kaoru1126 replied:
リダイレクトを追加しました。

seoky0 commented on app/views/messages/index.html.haml:
```diff
@@ -3,7 +3,8 @@
       %li.uname 16px bold #FFFFFF
-      %li.icons <i class="fa fa-cog" aria-hidden="true"></i> <i class="fa fa-pencil-square-o" aria-hidden="true"></i>
+      = link_to '', 'XXX', class:'fa fa-pencil-square-o'
+      = link_to '', "/users/#{current_user.id}/edit", class:'fa fa-cog'
```
パスはプレフィックスで書きましょう

Kaoru1126 replied:
修正しました。

seoky0 commented on config/routes.rb:
```diff
@@ -1,5 +1,10 @@
+   # User情報編集機能 no.6 ルートの設定？
+    resources :users, only: [:edit, :update]
+
+   # get '' => '#edit'
```
消しましょう

Kaoru1126 replied:
#6と#９を消しました。

seoky0 commented on test/controllers/users_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
消しましょう

Kaoru1126 replied:
ファイルごとけしました。

Kaoru1126 committed:
deleted unnecessary files

Kaoru1126 committed:
added resources to "messages"

Kaoru1126 committed:
changing the line to use prefix line

Kaoru1126 committed:
added before action and method for the action

Kaoru1126 committed:
deleted unnecessary lines and redirect method

Kaoru1126 committed:
fix certain prob

could not go to sign-in page and authenticate_user did not work well
=> deleted “except [index] or something”

this controller affects on every controller, so be careful about
writing something on this.

Kaoru1126 committed:
deleted unnecessary lines

Kaoru1126 committed:
moved the file and changed its name

Kaoru1126 committed:
change the order

this is for sync prob of variables

Kaoru1126 committed:
add space at the end

Kaoru1126 committed:
set strong params

Kaoru1126 committed:
added lines to give the params

yusukeyamane commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,33 @@
+
+  def update
+    if @user == current_user
+       @user.update(name:user_params[:name], email:user_params[:email])
```
@user.update(user_params)でいけませんか？

Kaoru1126 replied:
いけました。ありがとうございます！

yusukeyamane commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,33 @@
+       @user.update(name:user_params[:name], email:user_params[:email])
+       redirect_to controller: :messages, action: :index
+      else
+      redirect_to edit_user_path(current_user)
```
renderを使用しましょう！

Kaoru1126 replied:
修正済みです

yusukeyamane commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,33 @@
+       @user.update(name:user_params[:name], email:user_params[:email])
+       redirect_to controller: :messages, action: :index
+      else
+      redirect_to edit_user_path(current_user)
```
インデントを１つ浅くしましょう

Kaoru1126 replied:
修正済みです。

yusukeyamane commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,33 @@
+  end
+
+  private
+  def making_user
```
userを作成していると読める英語ですが、ここでは見つける、もしくはセットしているだけ。ですね。

Kaoru1126 replied:
"def set_user"に変更しました。

yusukeyamane commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,33 @@
+     @user = User.find(params[:id])
+  end
+
+  def configure_permitted_parameters
```
こちらはapplication controllerに書きましょう

Kaoru1126 replied:
修正しました。

yusukeyamane commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,33 @@
+  end
+
+  def move_to_sign_in
+    redirect_to new_user_session unless user_signed_in?
```
こちらはdeviseのデフォルトのメソッドを使用しましょう。

Kaoru1126 replied:
メソッドの中身を"user_session"に変えました。

Kaoru1126 committed:
few changed by the comment i got

1) def name for “making_user”
2)paramater’s name
3) redirect to render
4) in the def move to sign in’s way to call the method/
5)omitted the def “configure…” to the application contr…
